### PR TITLE
implements metadata for customer register and update

### PR DESCRIPTION
**Pull request by believelody**
### Your checklist for this pull request
Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.

🚨Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository.

- [ ] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *our develop*.
- [ ] Make sure you are requesting to pull request from a **topic/feature/bugfix branch** (right side). Don't pull request from your master!

What does this implement/fix? Explain your changes.
---------------------------------------------------
Allow Customer to include metadata in registerCustomer mutation and updateCustomer mutation

Does this close any currently open issues?
------------------------------------------
No

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
Metadata in registerCustomer mutation
image

Metadata in customers query
image

Metadata in updateCustomer mutation
image

Any other comments?
-------------------
Tell me if this enough and meet your requirements. Inspired by woocommerce-rest-api customer register feature

Where has this been tested?
---------------------------
**Operating System:** Ubuntu 18.04.5

**WordPress Version:** 5.5.3

### CODE REVIEW

**Comment by PaperPlaneSoftware** [comment]
Looks like the linting issues that this PR is failing on aren't related to the changes in this PR, is that correct? I *really* need this feature, any chance of merging it? I'm happy to help in any way I can to speed this up?
Cheers

**Comment by believelody** [comment]
@PaperPlaneSoftware feel free to check whatever you can add. I'm still waiting owners to review my PR. Hope they' will find some time, this library is so much great !

**Comment by aresrioja10** [comment]
Hi! I also need this feature! I'm using custom fields (save as meta data in Worpdress) in the billing and shipping form and it's the only way to update this information!

@believelody Can we help you to speed this up?

Thanks for your work by the way 🚀

**Comment by believelody** [comment]
@aresrioja10 sure, feel free to check what you can do. It seems the big problem is lint code. Still waiting for @kidunot89 to check if something is wrong.

**Review by kidunot89 — CHANGES_REQUESTED** [review]
@believelody @aresrioja10 I'm sorry for the delayed response been fixing a few fires with CI here and there. To answer your question the hold up is **testing**.

I can't add new code without ensuring its covered by the tests, too much risk involved with sites in production. This typically means for every new PR I have to usually pull down the PR branch, add a new test if the incoming changes warrant it and this change definitely warrants a test. I have two other PRs to do before this one, but if one of you adds a working test for the changes to this [test file](https://github.com/believelody/wp-graphql-woocommerce/blob/feature/customer-metadata/tests/wpunit/CustomerMutationsTest.php#L3). I can expedite this PR immediately.

File: tests/wpunit/CustomerMutationsTest.php [hunk 1]
```diff
@@ -563,4 +563,168 @@ public function testRegisterMutationWithoutAnyInfo() {
...
+								'value' => 'test_meta_value',
+							)
+						)
+					),
+				),
+			),
+		);
+
+		$this->assertEquals( $expected, $actual );
+	}
```
Comment (kidunot89) [inline tests/wpunit/CustomerMutationsTest.php:729]: @believelody @aresrioja10 Test added. Please use this as a reference in future when making PRs introducing logic that needs to be tested.

**Review by kidunot89 — APPROVED** [review]
@believelody @aresrioja10 With the test I added, this PR is ready for merging as soon as CI passes.

File: tests/wpunit/CustomerMutationsTest.php (hunk 1 above)
Comment (believelody) [reply tests/wpunit/CustomerMutationsTest.php:735]: Thank you so much @kidunot89.Don't worry for your delayed reply, I will use your test code and learn better about that.

Comment (aresrioja10) [reply tests/wpunit/CustomerMutationsTest.php:735]: @kidunot89 Thank you so much! I'm gonna test it right now :)

**Pull request merged**